products/BellHybrid/apps/application-bell-focus-timer/windows/FocusTimerWindow.hpp

```diff
@@ -17,10 +22,34 @@ namespace app::focus
                          std::unique_ptr<FocusTimerContract::Presenter> &&windowPresenter,
                          const std::string &name = window::name::timer);
 
+        void onBeforeShow(gui::ShowMode mode, gui::SwitchData *data) override;
         void buildInterface() override;
         bool onInput(const gui::InputEvent &inputEvent) override;
+        void onAllFocusSessionsFinished() override;
+        void onFocusSessionStarted() override;
+        void onFocusSessionFinished() override;
+        void onShortBreakStarted() override;
+        void onShortBreakFinished() override;
+        void pause() override;
+        void resume() override;
 
       private:
         std::unique_ptr<FocusTimerContract::Presenter> presenter;
+        gui::VBox *mainVBox                   = nullptr;
```

Comment:
Here I'd use
```suggestion
        gui::VBox *mainVBox{nullptr};
```
but it is up to you ;)